src/CreditCardForm.js

```diff
@@ -284,20 +299,56 @@ export default class CreditCardForm extends Component {
     const onError = this.onError
     const onComplete = this.onComplete
     const onValidationError = this.onValidationError
-    account.saveWithSecureForm(
-      this.props.publicKey,
-      form,
-      {
-        onError,
-        onComplete,
-        onNext,
-        onValidationError,
-      })
+
+    if (!!getToken !== false && typeof (getToken) === 'function') {
+      // call reauthorization method when able 
+      getToken(account.accountId)
+        .then(token => {
+          instrument.saveWithSecureForm(
+            token,
+            form,
+            {
+              onError,
+              onComplete,
+              onNext,
+              onValidationError,
+            })
+        })
+        .catch(error => console.error("Unable to save the instrument " + error))
+    } else {
+      // can use a public API key
+      if(!!publicKey === true){
+        instrument.saveWithSecureForm(
+          publicKey,
+          form,
+          {
+            onError,
+            onComplete,
+            onNext,
+            onValidationError,
+          })
+      } else if (!!accessToken === true) { // or pass accessToken separately
+        instrument.saveWithSecureForm(
+          accessToken,
+          form,
+          {
+            onError,
+            onComplete,
+            onNext,
+            onValidationError,
+          })
+      }
+    }
```

Comment:
This code block could use a refactor because the only different between both is `accessToken` vs `publicKey`.